src/cumulativeSum.js

```diff
@@ -0,0 +1,46 @@
+'use strict';
+
+import {Tensor, sizeOfShape} from './lib/tensor.js';
+import {validateCumulativeSumParams} from './lib/validate-input.js';
+
+/**
+ * Computes the cumulative sum of the input tensor along the specified axis.
+ * @param {Tensor} input
+ * @param {number} axis
+ * @param {MLCumulativeSumOptions} options
+ * @return {Tensor}
+ */
+export function cumulativeSum(input, axis, {exclusive = false, reverse = false} = {}) {
+  validateCumulativeSumParams(...arguments);
+  const inputShape = input.shape;
+  const outputShape = [...inputShape];
+  const output = new Tensor(outputShape);
+  const numElementsAlongAxis = inputShape[axis];
+
+  const totalElements = sizeOfShape(outputShape);
+
+  for (let outputIndex = 0; outputIndex < totalElements; outputIndex++) {
```

Comment:
It appears this nested `for` loop repeats the same computation multiple times for the same output axis, overwriting the same elements. I recommend skipping it if it's already computed in an earlier loop. e.g.

```
const inputElementStart = reverse ? elementCountAlongAxis - 1 : 0;
const inputElementStep = reverse ? -1 : 1;

for (let outputIndex = 0; outputIndex < totalElements; ++outputIndex) {
  const location = output.locationFromIndex(outputIndex);
  if (location[axis] >= 0) {
    continue; // No need to compute this axis again, since it was already done.
  }
  let cumulativeSumValue = 0;
  let inputLocation = [...location];
  let outputLocation = [...location];

  // Compute the accumulated sum along this entire axis.
  for (let i = 0; i < elementCountAlongAxis; ++i) {
    inputLocation[axis] = inputElementStart + i * inputElementStep;
    outputLocation[axis] = i;

    const inputValue = input.getValueByLocation(inputLocation);
    const oldCumulativeSumValue = cumulativeSumValue;
    cumulativeSumValue += inputValue;
    const outputValue = exclusive ? oldCumulativeSumValue : cumulativeSumValue;
    output.setValueByLocation(outputLocation, outputValue);
  }
}
```

Reply:
done

Reply:
Thanks for your advice! I missed this comment and have revised it now.